MVP Module/Source/Feed Presentation/FeedErrorViewModel.swift

```
@@ -0,0 +1,13 @@
+//
+//  FeedErrorViewModel.swift
+//  MVP
+//
+//  Created by Mher Movsisyan on 10.09.21.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+
+struct FeedErrorViewModel {
+	let failureReason: String
```

The `failureReason` could be an optional `String?`, so the Presenter can decide when to show/hide the message.

For example, in the `didStartLoadingFeed()` method, the Presenter can call `feedErrorView.display(FeedErrorViewModel(failureReason: .none))` to hide the error.

This way, the ViewController doesn't need to decide when to hide the error. It's easier to manage presentation state transitions when all changes are centralized in the presenter.

---

Thanks for suggestions